### PR TITLE
DOC added docs on using conda-forge w/ external MPI distributions

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds docs on using conda-forge with external MPI distributions. 

cc @leofang 

Also, I checked the ABI lab and openmpi is doing a pretty good job at ABI compat recently, so we might try and add the external builds for that too.

### CODE REVIEW

File: src/user/tipsandtricks.rst [hunk 1]
```diff
@@ -70,3 +70,35 @@ Using envs make it easier to debug problems with packages and ensure the stabili
...
+Using ``conda-forge`` w/ External Message Passing Interface (MPI) Libraries
+===========================================================================
+
+On some high-performance computing (HPC) systems, users are expected to use the
+MPI binaries that are available on the system as opposed to those built by ``conda-forge``.
+These binaries are typically specialized for the system and interface properly with queueing
+software, etc. However, this practice creates issues for ``conda-forge`` users. When you install
+a package from ``conda-forge`` that relies on MPI, ``conda`` will install the MPI binaries
+built by ``conda-forge`` and the package will link to those binaries. This setup tends to break
+or not function in unexpected ways on HPC systems.
```
Comment (leofang) [inline src/user/tipsandtricks.rst:84]:
```suggestion
built by ``conda-forge`` and the package will link to those binaries. This setup often either
does not work at all or functions in unexpected ways on HPC systems.
```

File: src/user/tipsandtricks.rst [hunk 2]
```diff
@@ -70,3 +70,35 @@ Using envs make it easier to debug problems with packages and ensure the stabili
...
+
+To solve these issues, ``conda-forge`` has created special dummy builds of the ``mpich`` libraries
+that are simply shell packages with no contents. These packages allow the ``conda`` solver to produce
+correct environments while eliminating the installed MPI binaries from ``conda-forge``. You can install the
+dummy package with the following command
```
Comment (leofang) [inline src/user/tipsandtricks.rst:89]:
```suggestion
correct environments while avoiding installing MPI binaries from ``conda-forge``. You can install the
dummy package with the following command
```

**Comment by isuruf** [comment]
> Also, I checked the ABI lab and openmpi is doing a pretty good job at ABI compat recently, so we might try and add the external builds for that too.

No, the 3.x and 4.x series are incompatible with each other and they are both supported at the moment.

**Comment by beckermr** [comment]
Right @isuruf but conda forge is pinned globally at 4 so we could provide a shim for 4 and tell ppl not to use it with 3. 

Within 4 things are liking ok.

File: src/user/tipsandtricks.rst [hunk 3]
```diff
@@ -70,3 +70,35 @@ Using envs make it easier to debug problems with packages and ensure the stabili
...
+does not work at all or functions in unexpected ways on HPC systems.
+
+To solve these issues, ``conda-forge`` has created special dummy builds of the ``mpich`` libraries
+that are simply shell packages with no contents. These packages allow the ``conda`` solver to produce
+correct environments while avoiding installing MPI binaries from ``conda-forge``. You can install the
+dummy package with the following command
+
+.. code-block:: shell
+
+    $ conda install mpich=*=external_*
```
Comment (isuruf) [inline src/user/tipsandtricks.rst:93]:
```suggestion
    $ conda install mpich=3.3.*=external_*
```

**Pull request merged**